cpp/arcticdb/version/version_utils.hpp

```diff
@@ -160,6 +160,29 @@ inline void check_is_version(const AtomKey &key) {
     util::check(key.type() == KeyType::VERSION, "Expected version key type but got {}", key);
 }
 
+inline void verify_stream_id(const StreamId &stream_id) {
+    if (ConfigsMap::instance()->get_int("VersionStore.NoStrictSymbolCheck"))
+    {
+        ARCTICDB_DEBUG(log::version(), "Key with stream id {} will not be strictly checked because VersionStore.NoStrictSymbolCheck variable is set to 1.", stream_id);
+        return;
+    }
+
+    if (std::holds_alternative<NumericId>(stream_id))
```

Comment:
Minor: Slightly prefer `util::variant_match` for the variant handling so that we get a compilation error if `StreamId` expands to include another type. That seems vanishingly unlikely for this case though, so fine to leave as-is, up to you.